Packs/Tenable_io/Integrations/TenableioEventCollector/TenableioEventCollector.py

```diff
@@ -0,0 +1,419 @@
+    Raises exceptions if something goes wrong.
+
+    :type client: ``Client``
+    :param Client: client to use
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+
+    client.get_audit_logs_request(limit=10)
+    return 'ok'
+
+
+''' MAIN FUNCTION '''
+
+
+def main() -> None:  # pragma: no cover
+    """main function, parses params and runs command functions
+    """
+    args = demisto.args()
+    command = demisto.command()
+    params = demisto.params()
+    events = []
+    vulnerabilities = []
+
+    access_key = params.get('access_key', {}).get('password')
+    secret_key = params.get('secret_key', {}).get('password')
+    vuln_fetch_interval = arg_to_number(params.get('vuln_fetch_interval', 240)) * 60  # type: ignore
+    severity = argToList(params.get('severity'))
+
+    verify_certificate = not params.get('insecure', False)
+    proxy = params.get('proxy', False)
+    max_fetch = arg_to_number(params.get('max_fetch')) or 1000
+
+    # transform minutes to seconds
+    first_fetch = arg_to_datetime(params.get('first_fetch'))
+
+    demisto.debug(f'Command being called is {command}')
+    try:
+        headers = {'X-ApiKeys': f'accessKey={access_key}; secretKey={secret_key}',
+                   "Accept": "application/json"}
+        client = Client(
+            verify=verify_certificate,
+            headers=headers,
+            proxy=proxy)
+
+        if command == 'test-module':
+            return_results(test_module(client))
+
+        elif command in ('tenable-get-audit-logs', 'tenable-get-vulnerabilities', 'fetch-events'):
+
+            if command == 'tenable-get-audit-logs':
+                results, events = get_audit_logs_command(client,
+                                                         from_date=args.get('from_date'),
+                                                         to_date=args.get('to_date'),
+                                                         actor_id=args.get('actor_id'),
+                                                         target_id=args.get('target_id'),
+                                                         limit=args.get('limit'))
+                return_results(results)
+            if command == 'tenable-get-vulnerabilities':
+                results, vulnerabilities = get_vulnerabilities_command(client,
+                                                                       last_found=args.get('last_found'),
+                                                                       num_assets=arg_to_number(args.get('num_assets')) or 5000,
+                                                                       severity=argToList(args.get('severity')))
+                return_results(results)
+            else:  # command == 'fetch-events':
+                last_run = demisto.getLastRun()
+                new_last_run_vuln = {}
+                if run_vulnerabilities_fetch(last_run=last_run, first_fetch=first_fetch,
```

Comment:
when this is false, new_last_run_vuln will be {} and next_fetch_vunl won't be stored for the next runs